internal/dag/gatewayapi_processor.go

```diff
@@ -64,11 +66,17 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 		// are associated with the Gateway. An empty Selector matches all routes.
 		for _, listener := range p.source.gateway.Spec.Listeners {
 
-			matches, err := selectorMatches(listener.Routes.Selector, route.Labels)
+			nsMatches, err := p.namespaceMatches(listener.Routes.Namespaces, route)
+			if err != nil {
+				p.Errorf("error validating namespaces against Listener.Routes.Namespaces: %s", err)
+			}
+
+			selMatches, err := selectorMatches(listener.Routes.Selector, route.Labels)
 			if err != nil {
 				p.Errorf("error validating routes against Listener.Routes.Selector: %s", err)
 			}
-			if matches {
+
+			if selMatches && nsMatches {
 				// Empty Selector matches all routes.
 				validRoutes = append(validRoutes, route)
```

Comment:
nit: `matchingRoutes` would be a more accurate name for this slice than `validRoutes`, I think.

Reply:
👍